### ISSUE TITLE
Would not advise to use this template!

### ISSUE THREAD

**Issue by Fusseldieb** [SimulatedGREG/electron-vue#1027]
Massively outdated depencies, build fails, author that doesn't fix, _respond_ or even accept pull requests. 
The right thing in this kind of situation would be archiving this repository and editing the README to include "OUTDATED". A 2 min task.

If the author of this repository sees this issue, please know that I'm not discouraging or even hating on someones project that probably took several hours, days or even weeks to complete, but in this situation it just does more harm than good.

But anyways, thanks for your contribution to open source code!

**Comment by mxswat** [comment]
Do you have any suggestions for alternatives?

**Comment by ikuokuo** [comment]
> Do you have any suggestions for alternatives?

Could use [Vue CLI Plugin Electron Builder](https://github.com/nklayman/vue-cli-plugin-electron-builder) instead.

Here is an article about how to create [Electron App with Vue.js, Vuetify](https://github.com/ikuokuo/start-electron/blob/master/docs/tutorial/5_electron-app.md).

**Comment by mxswat** [comment]
Thanks!

**Comment by mxswat** [comment]
> > Do you have any suggestions for alternatives?
> 
> Could use [Vue CLI Plugin Electron Builder](https://github.com/nklayman/vue-cli-plugin-electron-builder) instead.
> 
> Here are some articles about how to create an Electron App with Vue.js:
> 
> * [Electron App with Vue.js, Vuetify](https://github.com/ikuokuo/start-electron/blob/master/docs/tutorial/5_electron-app.md)
> * [Create an Electron application with Vue.js and Vuetify](https://medium.com/@bromix/electron-application-with-vue-js-and-vuetify-f2a1f9c749b8)

I did but, when I open the built version of the app the screen remains empty

**Comment by ClarkAllen1556** [comment]
It seems really odd that this template has died off so suddenly. It's pretty much the first result if you search "electron with vuejs" and it has the most stars and best document of the templates. What doesn't @SimulatedGREG seem to want to encourage an active community to support this template. It has over 12k stars and 1k forks, seems like someone out there would want to help. With over 200 issues open and most of them trying to get the dependencies updated and fix the initial setup though it seems hopeless.

**Comment by mxswat** [comment]
> > > Do you have any suggestions for alternatives?
> > 
> > 
> > Could use [Vue CLI Plugin Electron Builder](https://github.com/nklayman/vue-cli-plugin-electron-builder) instead.
> > Here are some articles about how to create an Electron App with Vue.js:
> > 
> > * [Electron App with Vue.js, Vuetify](https://github.com/ikuokuo/start-electron/blob/master/docs/tutorial/5_electron-app.md)
> > * [Create an Electron application with Vue.js and Vuetify](https://medium.com/@bromix/electron-application-with-vue-js-and-vuetify-f2a1f9c749b8)
> 
> I did but, when I open the built version of the app the screen remains empty

Anyway, the tutorial was fine but It didn't specify to use Hash mode in the router

**Comment by Fusseldieb** [comment]
> It seems really odd that this template has died off so suddenly. It's pretty much the first result if you search "electron with vuejs" and it has the most stars and best document of the templates. What doesn't @SimulatedGREG seem to want to encourage an active community to support this template. It has over 12k stars and 1k forks, seems like someone out there would want to help. With over 200 issues open and most of them trying to get the dependencies updated and fix the initial setup though it seems hopeless.

What I personally find rather odd that he's actively working on other stuff as it seems and not even saying a single "Hi, am busy" or something. 
I'm almost sure he can give another person the ownership, or at least permission to fork things, which would save this project.

**Comment by bromix** [comment]
Hi there, I'm the author of [Create an Electron application with Vue and Vuetify](https://medium.com/@bromix/electron-application-with-vue-js-and-vuetify-f2a1f9c749b8). I updated the article to work with the latest version of Electron 9 and the missing typings to work with TypeScript.

**Comment by 2227068304** [comment]
I wanted to help maintain this project but the author went missing

**Comment by DevJMD** [comment]
Just ran in to this issue myself. After seeing how outdated Electron was and the babel dependencies being outdated by three years I thought I'd try update them as there were so many NPM security issues (and the Electron build was massively outdated anyway).

In other news, this is a reminder to myself why I try not to depend so heavily on projects like this because it can be the life and difference between a maintainer choosing to abandoning a project or staying committed to it and that alone can come at a heavy cost of someone's business if the ethical choice was to trust the maintainer at the time. Unfortunately, it's always going to be a gamble because you don't know the maintainer's circumstances and it could well be out of their control for all we know. It's just a shame such a largely promoted repository appears to be dead in the water.

Sadly, it looks like I'm going to have to manually cherry pick every config and dependency (for example, Babel dropping preset support) until I get a stable build on Electron v10.0.0 and may even spin-off this repo as a maintained one myself from here on out as that seems the safest choice rather than going last-resort and scrapping it.

Edit: This was an existing project I took on, by the way. It was in desperate need of an update (multiple security vulns from dated software in NPM), and just days after writing this post, I decided in the end to scrap the entire Electron implementation and rebuild it with a clean, up-to-date version of Webpack, Babel, Vue and most importantly, Electron 10+ whilst trying my best to reuse the Vue components. 

Lessons taken from from this:

- Never depend on a single-person-driven project
- Always assume the maintainer may abandon the project
- If the principle of using such project relates to business-critical infrastructure:
    - Fork the project
    - Maintain the project as if it was your own
    - Restructure dependencies and reduce the need for such dependencies where absolutely possible
    - Contain the mentality that such project will live forever
    - Never, _ever_ depend on small, open-sourced projects if shareholders are a thing.
        - It's costly.
        - It's risky.
        - It's scary.

My advice: avoid this project. 

There's a great guide here that implements Vue, Vuex (for state management), Axios for HTTP requests  and of course, Electron itself.

Source: https://www.smashingmagazine.com/2020/07/desktop-apps-electron-vue-javascript/

**Comment by DevVue** [comment]
Yes same issue with outdated packages causing lots of issues, **as this post says i would advise not using this**. Thankfully i spotted these comments early on after facing issues, rather than battling through we decided to drop the project and start our own, that way we know the application.

**Comment by Nisthar** [comment]
I fucked up by using this package without checking if its maintained or not.

**Comment by genu** [comment]
> Hi there, I'm the author of [Create an Electron application with Vue and Vuetify](https://medium.com/@bromix/electron-application-with-vue-js-and-vuetify-f2a1f9c749b8). I updated the article to work with the latest version of Electron 9 and the missing typings to work with TypeScript.

Can't see the article due to Medium requiring registration. Maybe consider cross publishing on an open platform as well (like gist)

**Comment by bromix** [comment]
> > Hi there, I'm the author of [Create an Electron application with Vue and Vuetify](https://medium.com/@bromix/electron-application-with-vue-js-and-vuetify-f2a1f9c749b8). I updated the article to work with the latest version of Electron 9 and the missing typings to work with TypeScript.
> 
> Can't see the article due to Medium requiring registration. Maybe consider cross publishing on an open platform as well (like gist)

You don't need an account. Just use the incognito mode of your browser and you can read the article 😉

**Comment by maxiride** [comment]
Lucky me I stumbled upon this issue before committing myself to the project!

Sadly it doesn't seem there are any active forks worth mentioning: https://techgaun.github.io/active-forks/index.html#SimulatedGREG/electron-vue

**Comment by chongma** [comment]
thank you for this thread!  i nearly made the mistake of using this template.  this helped me https://github.com/ikuokuo/start-electron/blob/master/docs/tutorial/5_electron-app.md

**Comment by ValueEyecareNetwork** [comment]
> I fucked up by using this package

Same here :(

**Comment by loveStef** [comment]
> I fucked up by using this package

2022年的受害者来了

**Comment by jooy2** [comment]
I've split this into a new project with regards to it: https://github.com/jooy2/vutron

- Supports Vue 3, Electron 19~, NodeJS 14, 16~
- Supports the latest development environments such as Webpack 5, Babel 7, ESLint 8, etc.
- Faster compilation speed, various build, running problems fixed

I don't know if `electron-vue` will continue to be maintained, but I personally started working on new projects because I used this template often.

Please take a look around if you are interested.
